tools/generator/dfg/rp/rp_device_tree.py

```diff
@@ -0,0 +1,283 @@
+        return string
+
+    @staticmethod
+    def _device_tree_from_properties(p):
+        tree = DeviceTree('device')
+        tree.ids.append(p['id'])
+
+        def topLevelOrder(e):
+            order = ['attribute-flash', 'attribute-ram', 'attribute-eeprom', 'attribute-core', 'attribute-mcu', 'header', 'attribute-define']
+            if e.name in order:
+                if e.name in ['attribute-flash', 'attribute-eeprom', 'attribute-ram']:
+                    return (order.index(e.name), int(e['value']))
+                else:
+                    return (order.index(e.name), e['value'])
+            return (len(order), -1)
+        # tree.addSortKey(topLevelOrder)
+
+        # RPDeviceTree.addDeviceAttributesToNode(p, tree, 'attribute-flash')
+        # RPDeviceTree.addDeviceAttributesToNode(p, tree, 'attribute-ram')
+        # RPDeviceTree.addDeviceAttributesToNode(p, tree, 'attribute-eeprom')
+        # RPDeviceTree.addDeviceAttributesToNode(p, tree, 'attribute-mcu')
+
+        def driverOrder(e):
+            if e.name == 'driver':
+                if e['name'] == 'core':
+                    # place the core at the very beginning
+                    return ('aaaaaaa', e['type'])
+                if e['name'] == 'gpio':
+                    # place the gpio at the very end
+                    return ('zzzzzzz', e['type'])
+                # sort remaining drivers by type and compatible strings
+                return (e['name'], e['type'])
+            return ("", "")
+        tree.addSortKey(driverOrder)
+
+        # Core
+        core_child = tree.addChild('driver')
+        core_child.setAttributes('name', 'core', 'type', p['core'])
+        core_child.addSortKey(lambda e: (int(e['position']), e['name']) if e.name == 'vector' else (-1, ""))
+        core_child.addSortKey(lambda e: (e['name'], int(e['size'])) if e.name == 'memory' else ("", -1))
+
+        for section in p["memories"]:
+            memory_section = core_child.addChild("memory")
+            memory_section.setAttributes(["name", "access", "start", "size"], section)
+        # sort the node children by start address and size
+        core_child.addSortKey(lambda e: (int(e["start"], 16), int(e["size"])) if e.name == "memory" else (-1, -1))
+
+    
+        for vector in p['interrupts']:
+            if int(vector['position']) < 0: continue;
+            vector_section = core_child.addChild('vector')
+            vector_section.setAttributes(['position', 'name'], vector)
+
+        modules = {}
+        for m, i in p['modules']:
+            # filter out non-peripherals: fuses, micro-trace buffer
+            # gpio part
+            if m in ['pads_bank', 'pads_qspi']: continue;
+            # clock subsystem
+            if m in ['pll_usb', 'pll_sys', 'xosc']: continue;
+            if m not in modules:
+                modules[m] = [i]
+            else:
+                modules[m].append(i)
+
+
+        compatible = p['id']['platform'] 
```

Comment:
Let's use `rp20` here, so that if another rp is added later with incompatible peripherals we don't need to refactor modm (as happened already with some SAM devices).